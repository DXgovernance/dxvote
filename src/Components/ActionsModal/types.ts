import { DecodedAction } from 'old-components/Guilds/ActionsBuilder/types';

export interface ActionModalProps {
  isOpen: boolean;
  setIsOpen: (isOpen: boolean) => void;
  onAddAction: (action: DecodedAction) => void;
}
