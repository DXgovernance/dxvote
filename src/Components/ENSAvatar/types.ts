export interface ENSAvatarProps {
  address?: string;
  size?: number;
}
