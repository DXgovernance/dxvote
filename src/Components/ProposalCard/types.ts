import { ProposalStatusProps } from 'Components/ProposalStatus/types';
import { Proposal, ENSAvatar } from '../Types';

export interface ProposalCardProps {
  proposal?: Proposal;
  votes?: number[];
  ensAvatar?: ENSAvatar;
  href?: string;
  statusProps?: ProposalStatusProps;
}
