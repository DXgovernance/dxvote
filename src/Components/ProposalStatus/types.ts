import { Moment } from 'moment';
import { ProposalState } from 'types/types.guilds';

export interface ProposalStatusProps {
  endTime: Moment;
  status: ProposalState;
  bordered?: boolean;
  hideTime?: boolean;
  timeDetail?: string;
}
