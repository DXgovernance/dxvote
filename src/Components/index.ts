export { Header } from './Header';
export { WalletButton } from './WalletButton';
export { NetworkButton } from './NetworkButton';

