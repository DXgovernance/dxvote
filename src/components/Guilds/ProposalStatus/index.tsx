import { useMemo } from 'react';
import styled, { css } from 'styled-components';
import Skeleton from 'react-loading-skeleton';
import moment, { unix } from 'moment';

import { Box } from '../common/Layout';
import { Proposal, ProposalState } from '../../../types/types.guilds.d';

const Status = styled.div`
  font-size: 0.8rem;
  font-weight: 600;
  display: flex;
  flex-direction: row;
  align-items: center;
  ${({ bordered }) =>
    bordered &&
    css`
      border: 1px solid ${({ theme }) => theme.colors.text};
      border-radius: ${({ theme }) => theme.radii.pill};
      padding-left: 0.5rem;
    `}
`;

const Pill = styled(Box)`
  display: inline-flex;
  justify-content: center;
  align-items: center;

  border-radius: 1.5rem;
  padding: ${props => (props.padded ? '0.5rem 0.8rem' : '0')};
  color: ${props =>
    props.filled ? props.theme.colors.background : props.theme.colors.text};
  background-color: ${props =>
    props.filled ? props.theme.colors.text : 'transparent'};
  border: 1px solid ${({ theme }) => theme.colors.text};
`;

const DetailText = styled(Box)`
  padding: 0 0.2rem;

  @media only screen and (min-width: 768px) {
    padding-right: 0.5rem;
  }
`;

interface ProposalStatusProps {
  proposal: Proposal;
  bordered?: boolean;
  hideTime?: boolean;
  showRemainingTime?: boolean;
}

const ProposalStatus: React.FC<ProposalStatusProps> = ({
  bordered,
  proposal,
  hideTime,
  showRemainingTime,
}) => {
  const endTime = useMemo(() => {
    if (!proposal) return null;
    return unix(proposal.endTime.toNumber());
  }, [proposal]);

  const timeDetail = useMemo(() => {
    if (!endTime || hideTime) return null;

    const currentTime = moment();
    if (endTime.isBefore(currentTime) || showRemainingTime) {
      return endTime.fromNow();
    } else {
      return endTime.toNow();
    }
  }, [endTime, showRemainingTime, hideTime]);

  const statusDetail = useMemo(() => {
    if (!proposal || !endTime) return null;

    if (proposal.state === ProposalState.Submitted) {
      const currentTime = moment();
      if (endTime.isBefore(currentTime)) {
        return 'Ended';
      } else {
        return 'Active';
      }
    }

    return proposal.state;
  }, [endTime, proposal]);

  return (
    <Status bordered={hideTime ? false : bordered}>
      {!hideTime && (
        <DetailText>
          {endTime && timeDetail ? (
            <span title={endTime?.format('MMMM Do, YYYY - h:mm a')}>
              {timeDetail}
            </span>
          ) : (
            <Skeleton width={50} />
          )}
        </DetailText>
      )}
      <Pill filled padded>
        {statusDetail || (
          <Skeleton width={50} baseColor="#333" highlightColor="#555" />
        )}
      </Pill>
    </Status>
  );
};

export default ProposalStatus;
