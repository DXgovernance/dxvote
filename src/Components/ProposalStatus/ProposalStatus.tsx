import styled, { css } from 'styled-components';
import { Box } from 'Components/Primitives/Layout';
import { Loading } from 'Components/Primitives/Loading';
import { ProposalState } from 'Components/Types';
import { ProposalStatusProps } from './types';

const ProposalStatusWrapper = styled.div`
  display: flex;
  flex: 1;
  justify-content: flex-end;
`;

const Status = styled.div`
  font-size: 0.8rem;
  font-weight: 600;
  display: flex;
  flex-direction: row;
  align-items: center;
  ${({ bordered }) =>
    bordered &&
    css`
      border: 1px solid ${({ theme }) => theme.colors.text};
      border-radius: ${({ theme }) => theme.radii.pill};
      padding-left: 0.5rem;
    `}
`;

const ProposalStatusDetail = styled(Box)`
  display: inline-flex;
  justify-content: center;
  align-items: center;
  margin: 0.5rem;
  border-radius: 15px;
  border: 1px solid
    ${props =>
      props.statusDetail === ProposalState.Failed ? '#D500F9' : '#1DE9B6'};
  background-color: ${({ theme }) => theme.colors.background};
  color: ${props =>
    props.statusDetail === ProposalState.Failed ? '#D500F9' : '#1DE9B6'};
  padding: 0.25rem 0.4rem;
`;

const DetailText = styled(Box)`
padding: 0 0.2rem;

@media only screen and (min-width: 768px) {
  padding - right: 0.5rem;
}
`;

const ProposalStatus: React.FC<ProposalStatusProps> = ({
  endTime,
  status,
  bordered,
  hideTime,
  timeDetail,
}) => {
  return (
    <ProposalStatusWrapper>
      <Status test-id="proposal-status" bordered={hideTime ? false : bordered}>
        {!hideTime && (
          <DetailText>
            {timeDetail ? (
              <span title={endTime?.format('MMMM Do, YYYY - h:mm a')}>
                {timeDetail}
              </span>
            ) : (
              <Loading
                test-id="skeleton"
                loading
                text
                skeletonProps={{ width: '50px' }}
              />
            )}
          </DetailText>
        )}
        {status ? (
          <ProposalStatusDetail statusDetail={status}>
            {' '}
            {status}
          </ProposalStatusDetail>
        ) : (
          <Loading
            test-id="skeleton"
            loading
            text
            skeletonProps={{ width: '50px' }}
          />
        )}
      </Status>
    </ProposalStatusWrapper>
  );
};

export default ProposalStatus;
