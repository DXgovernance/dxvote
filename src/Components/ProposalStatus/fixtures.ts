import { ProposalState } from 'Components/Types';
import moment from 'moment';
import { ProposalStatusProps } from './types';

export const proposalStatusMock: ProposalStatusProps = {
  endTime: moment(),
  status: ProposalState.Active,
  timeDetail: '1 minute ago',
};
