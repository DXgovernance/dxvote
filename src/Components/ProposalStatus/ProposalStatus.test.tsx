import ProposalStatus from './ProposalStatus';
import { render } from '../../utils/tests';
import { ProposalStatusProps } from './types';
import { ProposalState } from 'Components/Types';
import moment from 'moment';

const validProps: ProposalStatusProps = {
  timeDetail: 'Time',
  status: ProposalState.Active,
  endTime: moment('2022-05-09T08:00:00'),
};

const invalidProps: ProposalStatusProps = {
  timeDetail: null,
  status: null,
  endTime: null,
};

test('ProposalStatus votes', async () => {
  const { container } = render(<ProposalStatus {...validProps} />);

  expect(container).toMatchSnapshot();
});

test('ProposalStatus loading', async () => {
  const { container } = render(<ProposalStatus {...invalidProps} />);

  expect(container).toMatchSnapshot();
});
