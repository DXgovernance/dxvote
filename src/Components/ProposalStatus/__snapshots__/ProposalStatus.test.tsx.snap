// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProposalStatus loading 1`] = `
<div>
   
  <div
    class="sc-gKseQn dTFpnB"
  >
    <div
      class="sc-iBPTik gOtsle"
    >
      <div
        class="sc-bdfBQB sc-pGacB bmhbKs bjIKgW"
      >
        <div
          class="sc-eCstlR"
        >
          <span
            aria-busy="true"
            aria-live="polite"
          >
            <span
              class="react-loading-skeleton"
              style="width: 50px; --base-color: #333; --highlight-color: #555;"
            >
              ‌
            </span>
            <br />
          </span>
        </div>
      </div>
      <div
        class="sc-eCstlR"
      >
        <span
          aria-busy="true"
          aria-live="polite"
        >
          <span
            class="react-loading-skeleton"
            style="width: 50px; --base-color: #333; --highlight-color: #555;"
          >
            ‌
          </span>
          <br />
        </span>
      </div>
    </div>
  </div>
   
</div>
`;

exports[`ProposalStatus votes 1`] = `
<div>
   
  <div
    class="sc-gKseQn dTFpnB"
  >
    <div
      class="sc-iBPTik gOtsle"
    >
      <div
        class="sc-bdfBQB sc-pGacB bmhbKs bjIKgW"
      >
        <span
          title="May 9th, 2022 - 8:00 am"
        >
          Time
        </span>
      </div>
      <div
        class="sc-bdfBQB sc-fubCzh bmhbKs ktAnLd"
      >
         
        Active
      </div>
    </div>
  </div>
   
</div>
`;
